src/libs/actions/User.js

```diff
@@ -119,11 +134,48 @@ function setSecondaryLogin(login, password) {
     });
 }
 
+/**
+ * Validates a login given an accountID and validation code
+ *
+ * @param {Number} accountID
+ * @param {String} validateCode
+ */
+function validateLogin(accountID, validateCode) {
+    const isLoggedIn = !_.isEmpty(sessionAuthToken);
+    const redirectRoute = isLoggedIn ? ROUTES.getReportRoute(currentlyViewedReportID) : ROUTES.SIGNIN;
+    Onyx.merge(ONYXKEYS.ACCOUNT, CONST.DEFAULT_ACCOUNT_DATA);
+
+    API.ValidateEmail({
+        accountID,
+        validateCode,
+    }).then((response) => {
+        if (response.jsonCode === 200) {
+            const {email} = response;
+
+            // If the user is unauthenticated, generate a login for them with the returned authtoken and sign then in
```

Comment:
Out of date comment I think since we are redirecting to sign in with a success/failure message